astropy/cosmology/core.py

```diff
@@ -1060,10 +1055,11 @@ def comoving_distance(self, z):
         """
 
         from scipy.integrate import quad
+        inv_e = self.inv_efunc
         if not isiterable(z):
```

Comment:
Sure, that could be done.  But that is irrelevant for this pull, as it has no effect on speed issues.